### PR TITLE
Add metadata to tile_source.py and fix various typos

### PR DESCRIPTION
**Pull request by jamestiotio**
This PR fixes part of #11765, supersedes #11901, and also includes typo fixes for various files under the `examples` directory.

### CODE REVIEW

File: examples/howto/server_embed/flask_gunicorn_embed.py [hunk 1]
```diff
@@ -1,7 +1,7 @@
 try:
     import asyncio
 except ImportError:
-    raise RuntimeError("This example requries Python3 / asyncio")
+    raise RuntimeError("This example requires Python3 / asyncio")
```
Comment (bryevdv) [inline examples/howto/server_embed/flask_gunicorn_embed.py:4]: I think this `try/catch` could actually be removed entirely? Bokeh itself requires Python 3 nowadays, so `asyncio` will always be available.

File: examples/howto/ipywidgets/README.md [hunk 2]
```diff
@@ -1,9 +1,9 @@
 Examples in this directory show off bokeh's capability to integrate with Jupyer widgets
-outside of JupyerLab or classical Notebook environments. All examples are bokeh server
+outside of JupyterLab or classical Notebook environments. All examples are bokeh server
 examples, so you run them with `bokeh server app_name`, and then you have to navigate
 to `http://localhost:5006/app_name`:
 
-* `ipyvolume_camera.py` -- shows how to integrate a thridparty widget and manipulate it
+* `ipyvolume_camera.py` -- shows how to integrate a thirdparty widget and manipulate it
```
Comment (bryevdv) [inline examples/howto/ipywidgets/README.md:6]:
```suggestion
* `ipyvolume_camera.py` -- shows how to integrate a third-party widget and manipulate it
```

File: examples/plotting/file/tile_source.py [hunk 3]
```diff
@@ -1,3 +1,12 @@
+""" This example displays a CartoDB Positron base world map.
+
+.. bokeh-example-metadata::
+    :apis: bokeh.plotting.Figure.add_tile
+    :refs:  :ref:`userguide_geo` > :ref:`userguide_tools`
+    :keywords: map, plot
```
Comment (bryevdv) [inline examples/plotting/file/tile_source.py:6]:
```suggestion
    :keywords: map, geo, tiles
```

**Comment by bryevdv** [comment]
@jamestiotio Wow thanks for the very thorough updates, I only have a couple of small suggestions

File: examples/howto/server_embed/flask_gunicorn_embed.py (hunk 1 above)
Comment (jamestiotio) [reply examples/howto/server_embed/flask_gunicorn_embed.py]: Okay got it, implemented in a43a152.

File: examples/howto/ipywidgets/README.md (hunk 2 above)
Comment (jamestiotio) [reply examples/howto/ipywidgets/README.md]: Resolved in a43a152.

**Comment by jamestiotio** [comment]
Hi @bryevdv, I have added your suggestions to this PR accordingly. Do kindly look through the changes again. Thanks!

**Review by bryevdv — APPROVED** [review]

**Comment by bryevdv** [comment]
Thanks @jamestiotio !

**Pull request merged**